apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/common/AbstractTransformerV2.java

```diff
@@ -106,14 +106,45 @@ protected static List<HumanName> createHumanNameFrom(PatientInfo patientInfo) {
         givens = null;
       }
 
-      names.add(new HumanName().setFamily(patientInfo.getLastName()).setGiven(givens));
+      HumanName name =
+          new HumanName()
+              .setFamily(patientInfo.getLastName())
+              .setGiven(givens)
+              .setText(createNameText(patientInfo));
+
+      names.add(name);
     } else {
       names = null;
     }
 
     return names;
   }
 
+  private static String createNameText(PatientInfo patientInfo) {
+    List<String> nodeNames = new ArrayList<>();
+    List<String> nodeFormats = new ArrayList<>();
+
+    if (patientInfo.getFirstName() != null) {
```

Comment:
@KyRobbins we need to adjust this to also ignore if the name is empty

Reply:
we need to do this for both the nameText and the givens array.  So if the middle initial is the empty string, it should appear in the givens arrary or the nameText